packages/abi-utils/lib/arbitrary.ts

```diff
@@ -0,0 +1,315 @@
+          fc.oneof(
+            fc.constant("pure"),
+            fc.constant("view"),
+            fc.constant("nonpayable"),
+            fc.constant("payable")
+          ),
+          fc.boolean()
+        )
+        .map(([stateMutability, includeOptionals]) =>
+          !includeOptionals
+            ? { stateMutability }
+            : {
+                stateMutability,
+                payable: stateMutability === "payable",
+                constant:
+                  stateMutability === "view" || stateMutability === "pure"
+              }
+        )
+    )
+    .map(records => records.reduce((a, b) => ({ ...a, ...b }), {}))
+    .filter(entry => {
+      const { inputs, outputs = [] } = entry as Types.FunctionEntry;
+      // names that are not blank should be unique
+      const names = [...inputs, ...outputs]
+        .map(({ name }) => name)
+        .filter(name => name !== "");
+      return names.length === new Set(names).size;
+    });
+
+export const ReceiveEntry = () =>
+  fc.record({
+    type: fc.constant("receive"),
+    stateMutability: fc.constant("payable")
+  });
+
+export const FallbackEntry = () =>
+  fc
+    .tuple(
+      fc.record({
+        type: fc.constant("fallback")
+      }),
+      fc
+        .tuple(
+          fc.oneof(fc.constant("nonpayable"), fc.constant("payable")),
+          fc.boolean()
+        )
+        .map(([stateMutability, includeOptionals]) =>
+          !includeOptionals
+            ? { stateMutability }
+            : {
+                stateMutability,
+                payable: stateMutability === "payable",
+                constant:
+                  stateMutability === "view" || stateMutability === "pure"
+              }
+        )
+    )
+    .map(records => records.reduce((a, b) => ({ ...a, ...b }), {}));
+
+export const ConstructorEntry = () =>
+  fc
+    .tuple(
+      fc.record({
+        type: fc.constant("constructor"),
+        inputs: fc.array(Parameter())
+      }),
+      fc
+        .tuple(
+          fc.oneof(fc.constant("nonpayable"), fc.constant("payable")),
+          fc.boolean()
+        )
+        .map(([stateMutability, includeOptionals]) =>
+          !includeOptionals
+            ? { stateMutability }
+            : {
+                stateMutability,
+                payable: stateMutability === "payable",
+                constant:
+                  stateMutability === "view" || stateMutability === "pure"
+              }
+        )
+    )
+    .map(records => records.reduce((a, b) => ({ ...a, ...b }), {}));
+
+export const Abi = () =>
+  fc
+    .tuple(
+      ConstructorEntry(),
+      FallbackEntry(),
+      ReceiveEntry(),
+      fc.array(fc.oneof(FunctionEntry(), EventEntry()))
+    )
+    .chain(([constructor, fallback, receive, entries]) =>
+      fc.shuffledSubarray([constructor, fallback, receive, ...entries])
+    );
+
+namespace Numerics {
+  // 0 < n <= 32
+  export const Bytes = () => fc.nat(31).map(k => 32 - k);
+
+  // o < n <= 256
```

Comment:
You have an `o` instead of a `0`. :P  Also you might want to specify `0 < n <= 256, 8 | n`? :)